bin/foreman-node

```diff
@@ -57,19 +61,41 @@ rescue Exception => e
   exit 1
 end
 
-def plain_grains(minion)
-  # We have to get the grains from the cache, because the client
-  # is probably running 'state.highstate' right now.
+def get_grains_from_filecache(minion)
+  # Use the grains from the salt master's filesystem based cache
+  # This requires the following settings in /etc/salt/foreman.yaml:
+  # :filecache: true
+  # :cachedir: "/path/to/master/cache" (default: "/var/cache/salt/master")
+  # Also, the msgpack rubygem needs to be present
+  cachedir = SETTINGS[:cachedir] || "/var/cache/salt/master"
+  file = File.open("#{cachedir}/minions/#{minion}/data.p", "rb")
```

Comment:
Easier would be:
    
    content = File.read("#{cachedir}/minions/#{minion}/data.p")

Reply:
done